src/MathOptInterface.jl

```diff
@@ -70,6 +70,21 @@ Supported file types depend on the solver or standalone instance type.
 function writeinstance end
 
 """
+    readinstance!(m::AbstractInstance, filename::String)
+
+Read the file `filename` into the instance `m`. If `m` is non-empty, this may
+throw an error.
+
+Supported file types depend on the instance type.
+
+    readinstance!(m::AbstractInstance, src::AbstractInstance)
```

Comment:
Maybe we can use `Base.copy!` for this version? It's exactly a destructive copy.

Reply:
I'm not a fan of using the Base version if everything else in MOI is unexported (i.e. get).
What about
```julia
MathOptInterface.write(m::AbstractInstance, filename::String)
MathOptInterface.read!(m::AbstractInstance, filename::String)
MathOptInterface.copy!(m::AbstractInstance, src::AbstractInstance)
```

Reply:
Someone's going to ask why we're not consistent with Base, but yes, that's okay with me.

Reply:
Imagine if this worked...
```julia
mof = MOFInstance("test.mof.json")

grb = MOI.SolverInstance(GurobiSolver())
MOI.copy!(grb, mof)
MOI.write(grb, "test.mps")

cpx = MOI.SolverInstance(CPLEXSolver())
MOI.read!(cpx, "test.mps")

mof2 = MOFInstance()
MOI.copy!(mof2, cpx)

mof2 == mof
```

Reply:
Replace `==` with my `test_instances_equal` (github.com/JuliaOpt/MathOptInterfaceUtilities.jl/pull/22/files#diff-d96547c1b2a274f333396d4d54349d4aR263) and it should work.